Assets/MixedRealityToolkit/Utilities/AssetScriptReferenceRetargeter.cs

```diff
@@ -0,0 +1,593 @@
+                            line += await reader.ReadLineAsync();
+                            lineNum++;
+
+                            if (!line.Contains('}'))
+                            {
+                                throw new InvalidDataException($"Unexpected part of YAML line split over more than two lines, starting two lines: {line}");
+                            }
+                        }
+
+                        if (line.Contains(ScriptFileIdConstant))
+                        {
+                            Match regexResults = Regex.Match(line, @"guid:\s*([0-9a-fA-F]*)");
+                            if (!regexResults.Success || regexResults.Groups.Count != 2 || !regexResults.Groups[1].Success || regexResults.Groups[1].Captures.Count != 1)
+                            {
+                                throw new InvalidDataException($"Failed to find the guid in line: {line}.");
+                            }
+
+                            string guid = regexResults.Groups[1].Captures[0].Value;
+                            if (remapDictionary.TryGetValue(guid, out Tuple<string, long> tuple))
+                            {
+                                line = $"  m_Script: {{fileID: {tuple.Item2}, guid: {tuple.Item1}, type: 3}}";
+                            }
+                            else if (nonClassDictionary.ContainsKey(guid))
+                            {
+                                throw new InvalidDataException($"A script without a class ({nonClassDictionary[guid]}) is being processed.");
+                            }
+                            else
+                            {
+                                // Switch to error later
+                                Debug.LogWarning($"Couldn't find a script remap for {guid} in file: '{filePath}' at line '{lineNum}'.");
+                            }
+                        }
+                        // else this is not a script file reference
+                    }
+                    else if (line.Contains(ScriptFileIdConstant))
+                    {
+                        throw new InvalidDataException($"Line contains script type but not m_Script: {line}");
+                    }
+                    //{ fileID: 11500000, guid: 83d9acc7968244a8886f3af591305bcb, type: 3}
+
+                    await writer.WriteLineAsync(line);
+                }
+            }
+        }
+
+        private static bool IsYamlFile(string filePath)
+        {
+            string extension = Path.GetExtension(filePath);
+            if (ExcludedYamlAssetExtensions.Contains(extension))
+            {
+                return false;
+            }
+
+            using (StreamReader reader = new StreamReader(filePath))
+            {
+                return reader.ReadLine().StartsWith(YamlPrefix);
+            }
+        }
+
+        private static Dictionary<string, ClassInformation> ProcessScripts(string[] allFilePaths)
```

Comment:
>private static Dictionary<string, ClassInformation> ProcessScripts(string[] allFilePaths) [](start = 8, length = 89)

Nit: This is a general comment for a number of these functions which return dictionaries (especially because they return general types i.e. string), it would be helpful to have a comment saying what the semantics of the returned values are (i.e. "returns a dictionary of asset full assembly names to ClassInformtion values"